nixos/modules/services/blockchain/ethereum/geth.nix

```
@@ -17,6 +17,15 @@ let
         description = lib.mdDoc "Port number Go Ethereum will be listening on, both TCP and UDP.";
       };
 
+      ipc = {
+        enable = lib.mkEnableOption (lib.mdDoc "Enables the IPC-RCP server");
+        path = mkOption {
+          type = types.str;
```

Shouldn't this be a `types.path`? And shouldn't it default to `null`?

---

Thank for the review. I tried using types.path and setting the default to `null`.  But I got the following error when testing. 
```
│error: A definition for option `services.geth.test.ipc.path' is not of type `path'. Definition values: geth.nix: null
```
Also it looks like [`types.path`](https://github.com/NixOS/nixpkgs/blob/master/lib/types.nix#L482) doesn't allow relative paths. When Geth is given a relative path for ipcPath, it uses dataDir as root, which can be convenient.

---

Absolute paths in general are better. But lets say what others say.

---

```suggestion
          type = types.nullOr types.str;
```
types.path only works for absolute paths that are present at evaluation time. using type.str is fine here